receiver/googlecloudspannerreceiver/internal/filterfactory/itemfilterfactory.go

```diff
@@ -0,0 +1,102 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filterfactory
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudspannerreceiver/internal/filter"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudspannerreceiver/internal/metadata"
+)
+
+const (
+	defaultMetricDataPointsAmountInPeriod = 24 * 60
+	defaultItemActivityPeriod             = 24 * time.Hour
+)
+
+type itemFilterFactory struct {
+	filterByMetric map[string]filter.ItemFilter
+}
+
+type ItemFilterFactoryConfig struct {
+	MetadataItems  []*metadata.MetricsMetadata
+	TotalLimit     int
+	ProjectAmount  int
+	InstanceAmount int
+	DatabaseAmount int
+}
+
+func NewItemFilterFactory(logger *zap.Logger, config *ItemFilterFactoryConfig) (filter.ItemFilterResolver, error) {
```

Comment:
This is also another stutter, please consider renaming.

```suggestion
func NewItemResolver(logger *zap.Logger, config *ItemFilterFactoryConfig) (filter.ItemFilterResolver, error)
```

Reply:
I totally don't agree here, because it is not Item resolver, it is Item Filter Resolver. Current name reflects purpose.

Reply:
The current method implies it is gonna built a factory but returns a `ItemFilterResolver` so the naming is somewhat confusing here.

Reply:
Ok, Now I see what you've meant. Will update now.

Reply:
Done.